db/sql/prototype.sql

```diff
@@ -0,0 +1,491 @@
+
+
+CREATE OR REPLACE FUNCTION
+msar.change_table_name(table_id oid, new_name text) RETURNS text AS $$/*
+Change a table's name, returning the command executed.
+
+Args:
+  table_id:  the OID of the table whose name we want to change
+  new_name:  unquoted, unqualified table name
+*/
+BEGIN
+  RETURN __msar.change_table_name(__msar.get_table_name(table_id), quote_ident(new_name));
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+CREATE OR REPLACE FUNCTION
+msar.change_table_name(schema_ text, old_name text, new_name text) RETURNS text AS $$/*
+Change a table's name, returning the command executed.
+
+Args:
+  schem: unquoted schema name where the table lives
+  old_name:  unquoted, unqualified original table name
+  new_name:  unquoted, unqualified new table name
+*/
+DECLARE fullname text;
+BEGIN
+  fullname := msar.get_fq_table_name(schema_, old_name);
+  RETURN __msar.change_table_name(fullname, quote_ident(new_name));
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+-- Comment on table --------------------------------------------------------------------------------
+
+CREATE OR REPLACE FUNCTION
+__msar.comment_on_table(name_ text, comment_ text) RETURNS text AS $$/*
+Change the description of a table, returning command executed.
+
+Args:
+  name_: The qualified, quoted name of the table whose comment we will change.
+  comment_: The new comment. Any quotes or special characters must be escaped.
+*/
+BEGIN
+  RETURN __msar.exec_ddl('COMMENT ON TABLE %s IS ''%s''', name_, comment_);
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+CREATE OR REPLACE FUNCTION
+msar.comment_on_table(table_id oid, comment_ text) RETURNS text AS $$/*
+Change the description of a table, returning command executed.
+
+Args:
+  table_id: The OID of the table whose comment we will change.
+  comment_: The new comment. Any quotes or special characters must be escaped.
+*/
+BEGIN
+  RETURN __msar.comment_on_table(__msar.get_table_name(table_id), comment_);
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+CREATE OR REPLACE FUNCTION
+msar.comment_on_table(schema_ text, name_ text, comment_ text) RETURNS text AS $$/*
+Change the description of a table, returning command executed.
+
+Args:
+  schema_: The schema of the table whose comment we will change.
+  name_: The name of the table whose comment we will change.
+  comment_: The new comment. Any quotes or special characters must be escaped.
+*/
+DECLARE qualified_name text;
+BEGIN
+  qualified_name := msar.get_fq_table_name(schema_, name_);
+  RETURN __msar.comment_on_table(qualified_name, comment_);
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+-- Alter Table: LEFT IN PYTHON (for now) -----------------------------------------------------------
+
+-- Update table primary key sequence to latest -----------------------------------------------------
+
+CREATE OR REPLACE FUNCTION
+__msar.update_pk_sequence_to_latest(table_name text, column_ text) RETURNS text AS $$/*
```

Comment:
Is this compatible with multi-column PKs?

Reply:
No, this is a direct port of the current python function (which also doesn't handle multi-column primary keys).

My plan is to get the current logic ported over while making minimal changes. I'm hoping to use pretty much the same test suite.

However, reviewing all this is a great opportunity to write up bugs and issues about exactly this sort of thing.

I'm pretty sure (hoping) that this function is only called when we're already sure it's a single-column primary key, namely when we've set the default `id` column as `SERIAL PRIMARY KEY` after the splitting logic runs. If that's wrong, we definitely need a bug report issue for this.